docs/DesignDocs/oe_log_message()_callback_proposal.md

```diff
@@ -0,0 +1,61 @@
+Proposal of Public Callback API for oe_log_message()
+====
+
+# Motivation
+
+This document describes the proposed public callback API for oe_log_message().
+
+There are several problems about the current logging that prevent end users to
+make full use of log:
+- It does't have mechanism to communicate between enclave and host.
+- Message can't be redirected to different destinations on demand.
+- Lack of method for end user to customize messages.
+
+The callback API is supposed to solve such problems.
+
+# User Experience
+
+By adding this public API, user can customize the output of the log message:
+- The output destination can be stdout/stderr or any specified files, and it is
+adjustable during runtime.
+- The message can be adjusted according to the log level.
+- The user can get as much information about the message as internel API. For
+example, log time, enclave side/host side, log level.
+- Finally user can define a function to customize the log message, and can
+designate such funtion to process all the log message.
+
+# Specification
+
+Several neccessary items should be added to public header file.
+- Since log level is exposed to user, the related type should also be exposed:
+`
```

Comment:
When writing blocks of code, you should use the triple backticks instead, otherwise everything renders as a single line. Doing so will also allow you to specify the type of information it contains for syntax highlighting:  
````md
```c
typedef enum _oe_log_level
{
    OE_LOG_LEVEL_NONE = 0,
    OE_LOG_LEVEL_FATAL,
    OE_LOG_LEVEL_ERROR,
    OE_LOG_LEVEL_WARNING,
    OE_LOG_LEVEL_INFO,
    OE_LOG_LEVEL_VERBOSE,
    OE_LOG_LEVEL_MAX
} oe_log_level_t;
```
````
will render as:
```c
typedef enum _oe_log_level
{
    OE_LOG_LEVEL_NONE = 0,
    OE_LOG_LEVEL_FATAL,
    OE_LOG_LEVEL_ERROR,
    OE_LOG_LEVEL_WARNING,
    OE_LOG_LEVEL_INFO,
    OE_LOG_LEVEL_VERBOSE,
    OE_LOG_LEVEL_MAX
} oe_log_level_t;
```
instead of how it looks right now: `typedef enum _oe_log_level
{
    OE_LOG_LEVEL_NONE = 0,
    OE_LOG_LEVEL_FATAL,
    OE_LOG_LEVEL_ERROR,
    OE_LOG_LEVEL_WARNING,
    OE_LOG_LEVEL_INFO,
    OE_LOG_LEVEL_VERBOSE,
    OE_LOG_LEVEL_MAX
} oe_log_level_t;
`

Reply:
Updated, Thanks for the suggestion!